src/fswAlgorithms/sensorInterfaces/simpleInstrumentController/simpleInstrumentController.c

```diff
@@ -87,24 +88,26 @@ void Update_simpleInstrumentController(simpleInstrumentControllerConfig *configD
     sigma_BR_norm = v3Norm(attGuidInMsgBuffer.sigma_BR);
     omega_BR_norm = v3Norm(attGuidInMsgBuffer.omega_BR_B);
 
-    // If the target has not been imaged
-    if (!configData->imaged) {
-        /* If the attitude error is less than the tolerance, the groundLocation is accessible, and (if enabled) the rate
-        error is less than the tolerance, turn on the instrument and set the imaged indicator to 1*/
-        if ((sigma_BR_norm <= configData->attErrTolerance)
-            && (!configData->useRateTolerance || (omega_BR_norm <= configData->rateErrTolerance)) // Check rate tolerance if useRateTolerance enabled
-            && (accessInMsgBuffer.hasAccess))
-        {
-            deviceCmdOutMsgBuffer.deviceCmd = 1;
-            configData->imaged = 1;
-            // Otherwise, turn off the instrument
+    // If the controller is active
+    if (configData->controllerStatus) {
```

Comment:
Unless I'm mistaken you could remove this condition. You set the value to zero above with the init of the message to zero, yet the condition is not true you set it to 0 again. Similarly with the `else` branch in the innermost condition.

Reply:
If I understood you correctly, this should be resolved now. I removed the two else{} conditions that would set the command message to zero.